include/api.php

```diff
@@ -1485,6 +1485,71 @@ function api_users_search($type)
 /// @TODO move to top of file or somewhere better
 api_register_func('api/users/search', 'api_users_search');
 
+/**
+ * Returns statuses that match a specified query.
+ *
+ * @see https://developer.twitter.com/en/docs/tweets/search/api-reference/get-search-tweets
+ *
+ * @param string $type Return format: json, xml, atom, rss
+ *
+ * @return array|string
+ * @throws UnauthorizedException
+ * @throws BadRequestException
+ */
+function api_search($type)
+{
+	$data = array();
+
+	if (x($_REQUEST, 'q')) {
+		if (x($_REQUEST, 'rpp')) {
+			$count = $_REQUEST['rpp'];
+		} elseif (x($_REQUEST, 'count')) {
+			$count = $_REQUEST['count'];
+		} else {
+			$count = 15;
+		}
+
+		$since_id = (x($_REQUEST, 'since_id') ? $_REQUEST['since_id'] : 0);
+		$max_id = (x($_REQUEST, 'max_id') ? $_REQUEST['max_id'] : 0);
+		$page = (x($_REQUEST, 'page') ? $_REQUEST['page'] - 1 : 0);
+
+		$start = $page * $count;
+
+		if ($max_id > 0) {
+			$sql_extra .= ' AND `item`.`id` <= ' . intval($max_id);
+		}
+
+		$r = q(
+			"SELECT %s
+			FROM `item` %s
+			WHERE %s AND (`item`.`uid` = 0 OR (`item`.`uid` = %s AND NOT `item`.`global`))
+			AND `item`.`body` REGEXP '%s'
+			$sql_extra
+			AND `item`.`id`>%d
+			GROUP BY `item`.`uri`, `item`.`id`
```

Comment:
grouping by item.id doesn't seem to be a good idea - this field is unique, so grouping after it doesn't make sense. Additionally I think that you shouldn't need the grouping at all here.

Reply:
Grouping by a unique field can make sense when you’re doing a one-to-many join but only need a single row per unique value, usually when you need the result of an aggregate function like COUNT or MAX.

Reply:
Yeah, but this isn't the case here.

Reply:
On mobile, I can’t see the code past your comment, so I couldn’t see the SELECT fields! 😅